### ISSUE TITLE
Investigate validation loss increasing

### ISSUE THREAD

**Issue by marianovitasari20** [alan-turing-institute/icenet-mp#258]
### Problem / Goal

The validation loss is consistently increasing when using the current default `base` / `sample_dataset` configuration. This behavior needs to be investigated.

### What does good look like?

- [ ] Validation loss should show a downward trend during training and converge or stabilize, rather than consistently increasing.

**Comment by marianovitasari20** [comment]
I’m currently working on this.

**Comment by marianovitasari20** [comment]
Changing the loss function can affect training and validation loss behavior. For example, in fully deterministic mode for a fair comparison, I compared: 
(1) L1 (MAE) loss: https://wandb.ai/turing-seaice/train/runs/4zdg0p8i (the loss increases)
(2) MSE loss: https://wandb.ai/turing-seaice/train/runs/v0u37f1z (the loss decreases)
(3) Weighted MSE loss (with weights from the active grid cell mask): 

I need to run more experiments with additional epochs and different seeds before drawing conclusions. I haven’t finished yet. Isambard has just come back online, it was down yesterday.

**Comment by marianovitasari20** [comment]
1) L1 (MAE) loss

<img width="966" height="322" alt="Image" src="https://github.com/user-attachments/assets/829dce45-5af3-45ec-8885-3e09c966cb66" />

2) MSE loss

<img width="964" height="318" alt="Image" src="https://github.com/user-attachments/assets/a1f862ed-13a7-49c8-8622-23e3dcffa4d5" />

**Comment by marianovitasari20** [comment]
<img width="1055" height="312" alt="Image" src="https://github.com/user-attachments/assets/d2c567da-fb32-45ff-add4-362687081540" />

With fully deterministic mode (MAE loss), the validation loss decreases after reprojection instead of increasing. This does not happen for all seeds, so I need to run more tests with different seeds to confirm.

**Comment by marianovitasari20** [comment]
MAE vs MSE

<img width="1163" height="313" alt="Image" src="https://github.com/user-attachments/assets/fb58ebbc-3fb7-4d96-b266-ef2c0c59d976" />

<img width="1159" height="314" alt="Image" src="https://github.com/user-attachments/assets/8bf9a773-f192-42e8-a76a-6bcfc6b23003" />

**Comment by marianovitasari20** [comment]
seed: 555
<img width="1162" height="317" alt="Image" src="https://github.com/user-attachments/assets/db0f3bf9-4fd4-422b-8b0e-88353adaf796" />
<img width="1160" height="309" alt="Image" src="https://github.com/user-attachments/assets/c9175b49-9a75-46ae-b982-16b6bcc4dbc6" />
<img width="1163" height="315" alt="Image" src="https://github.com/user-attachments/assets/456606bc-901a-4a47-b137-dbbff1f81728" />

**Comment by marianovitasari20** [comment]
seed: 123
<img width="1160" height="317" alt="Image" src="https://github.com/user-attachments/assets/03a6d570-dce7-4bc7-a5bd-cbb6ac370e5b" />
<img width="1161" height="314" alt="Image" src="https://github.com/user-attachments/assets/f1fb9795-b81a-4ae3-bc45-723b573ee3ce" />
<img width="1162" height="317" alt="Image" src="https://github.com/user-attachments/assets/3927242b-c5af-47b5-b93f-1411f92026f3" />

**Comment by marianovitasari20** [comment]
seed: 731
<img width="1158" height="315" alt="Image" src="https://github.com/user-attachments/assets/e334d9c5-e178-4eda-977b-5b475a16d44e" />
<img width="1165" height="315" alt="Image" src="https://github.com/user-attachments/assets/0958aa46-898d-4edc-aa82-e839dd3b5f98" />
<img width="1158" height="315" alt="Image" src="https://github.com/user-attachments/assets/e8fcea6e-e02e-4adf-b191-82b7b67a33d5" />

**Comment by marianovitasari20** [comment]
seed: 482
<img width="1162" height="314" alt="Image" src="https://github.com/user-attachments/assets/eed3e59a-c8c9-4442-8c88-7193163c8550" />
<img width="1156" height="314" alt="Image" src="https://github.com/user-attachments/assets/e849204e-411f-4c79-a2de-4f559e141c12" />
<img width="1160" height="312" alt="Image" src="https://github.com/user-attachments/assets/b6217966-f916-442e-8903-478d5eb5c66c" />

**Comment by marianovitasari20** [comment]
seed: 294
<img width="1159" height="312" alt="Image" src="https://github.com/user-attachments/assets/227f7fd0-1509-4d22-9281-4f92505a8ef1" />
<img width="1159" height="312" alt="Image" src="https://github.com/user-attachments/assets/188b08de-e763-42b2-8156-6aad2b03d96e" />
<img width="1158" height="311" alt="Image" src="https://github.com/user-attachments/assets/d2bb9e72-1133-4521-bb8f-7b9f68caff00" />

**Comment by marianovitasari20** [comment]
MSE, without reprojection
<img width="1162" height="319" alt="Image" src="https://github.com/user-attachments/assets/30515502-2ec8-4d36-b3c7-67ccdb4a0abc" />

MSE, with reprojection
<img width="2318" height="628" alt="Image" src="https://github.com/user-attachments/assets/fe78a571-10cc-425c-850e-89a792cc0c1a" />

**Comment by marianovitasari20** [comment]
<img width="1160" height="315" alt="Image" src="https://github.com/user-attachments/assets/6e540b65-b678-465b-b5f4-88dcc6fd1807" />
<img width="1158" height="311" alt="Image" src="https://github.com/user-attachments/assets/92dcc8bd-c370-4455-8319-60994db1b8fd" />
<img width="1163" height="311" alt="Image" src="https://github.com/user-attachments/assets/70d0716c-f201-4c1a-825a-75970abb87d4" />
<img width="1161" height="314" alt="Image" src="https://github.com/user-attachments/assets/9283c204-d921-4128-90cd-dffd61d3fb43" />
<img width="1161" height="316" alt="Image" src="https://github.com/user-attachments/assets/61e1fb18-a25c-4267-91f5-c3f69741f43e" />

**Comment by marianovitasari20** [comment]
Still running more experiments. I'll write the analysis later once I finish running all the experiments.

**Comment by marianovitasari20** [comment]
MAE, without reprojection
<img width="1163" height="311" alt="Image" src="https://github.com/user-attachments/assets/829857e0-0b4c-4bcd-9077-820844b63c91" />

MAE, with reprojection
<img width="2326" height="626" alt="Image" src="https://github.com/user-attachments/assets/d05f0836-2760-40e8-8e70-acdb32738424" />

**Comment by marianovitasari20** [comment]
<img width="1160" height="312" alt="Image" src="https://github.com/user-attachments/assets/801c7d3a-5752-4a4b-8e31-7fe1338e2977" />

<img width="1161" height="315" alt="Image" src="https://github.com/user-attachments/assets/5b707da5-aa3d-4a49-bbe2-2e03324736af" />

<img width="1164" height="317" alt="Image" src="https://github.com/user-attachments/assets/49b59cf7-7ff8-4917-8e7b-c838679d0fe7" />

<img width="1160" height="314" alt="Image" src="https://github.com/user-attachments/assets/f0e3cb30-9675-4997-9d3a-7f5cf717af8e" />

<img width="1163" height="316" alt="Image" src="https://github.com/user-attachments/assets/e660e81c-407c-4406-bb92-549d8eaa8684" />

**Comment by marianovitasari20** [comment]
There are many insights we can take from these experiments. I’ll write up the analysis later.

**Comment by marianovitasari20** [comment]
**Analysis:**
**1. MAE vs. MSE:** MSE appears to perform better overall, and is more standard in deep learning due to its differentiability. I'll switch our current loss from MAE to MSE in a separate PR. Also, since the loss is not really configurable, I am working on making it configurable in the same PR.
**2. Reprojection:** This seems to help, the validation loss decreases rather than increases when reprojection is applied, especially in the early steps (before the loss begins increasing again, possibly due to overfitting).
**3. Loss increasing after initial decrease:** This could indicate overfitting. Worth monitoring and potentially revisiting regularization or early stopping. I'll track this in a new PR.
**4. Validation set dates:** I’m currently experimenting with different validation sets to see if this is causing the validation loss to increase. I encountered errors for certain dates (possibly because the data is unavailable, but I need to investigate further) and will look into this in a new PR.

**Comment by jemrobinson** [comment]
1. Most weather forecasting models are using MAE. Do you want to ask around why this is the case @marianovitasari20 ?
2. Good to know
3. Could be related to the small sample dataset. Worth running on the full dataset?
4. TBD

**Comment by louisavz** [comment]
2. It doesn't appear to be consistently the case (e.g., seed 555). Good to see more runs with reprojection appears to possibly overfit instead of completely not learning.  

The training loss consistently increases near epoch 13. Any thoughts as to why? Also, could we include the learning rate plot in future runs?

**Comment by marianovitasari20** [comment]
Based on today’s meeting with @IFenton and @jemrobinson, I will hold this PR until [https://github.com/alan-turing-institute/icenet-mp/issues/279](https://github.com/alan-turing-institute/icenet-mp/issues/279) and [https://github.com/alan-turing-institute/icenet-mp/issues/275](https://github.com/alan-turing-institute/icenet-mp/issues/275) are resolved.

I need to address both issues before continuing work on this PR. After that, I’m happy to run experiments on the full datasets.

I also think the current results (loss increasing after initial decrease) could be due to overfitting, or potentially because of the current validation set selection (sample), but I still cannot evaluate on other validation sets due to the error when changing them. Because of this, it makes sense to wait until we evaluate on the full datasets and on sample datasets with different validation sets.

**Comment by marianovitasari20** [comment]
Running with sample datasets to gather insights while waiting for the full datasets to become available (they are currently still downloading).

**Comment by marianovitasari20** [comment]
**Sample datasets, U-Net, Huber loss**
seed: 555
<img width="1440" height="316" alt="Image" src="https://github.com/user-attachments/assets/c0b6d1d6-2f01-4415-bf74-9693981c0046" />
seed: 123
<img width="1439" height="314" alt="Image" src="https://github.com/user-attachments/assets/1d00bbfa-ba3d-48dc-9508-c4ad1cddf6d0" />
seed: 294
<img width="1438" height="315" alt="Image" src="https://github.com/user-attachments/assets/cb54ed0e-6f2f-4d29-9808-70cb1d7f859d" />

**Comment by marianovitasari20** [comment]
<img width="2902" height="644" alt="Image" src="https://github.com/user-attachments/assets/123bdc15-d980-4295-ac98-38e469541599" />

I just finished training on the same sample dataset (not the full dataset) using DDPM (instead of U-Net) with 125 epochs, and as you can see, the val loss is much more stable. This is probably due to:

**UNet predicts in one shot**: if it overfits or the val distribution shifts slightly, the loss can spike back up. DDPM on the other hand samples over many denoising steps, which acts as a natural regularizer.

**Different training objectives**: UNet minimizes MSE directly on the output, making it more sensitive to train/val distribution mismatch. DDPM optimizes a denoising objective (predicting v-parameterization), which is a smoother training signal.

**Averaging over diffusion timesteps**: at each training step, DDPM randomly samples a timestep t and learns to denoise from that noise level. This averaging effect makes the loss less sensitive to any single batch, resulting in a smoother curve.

**Comment by marianovitasari20** [comment]
currently running with full dataset (south): https://wandb.ai/turing-seaice/train/runs/k4ep3xba and https://wandb.ai/turing-seaice/train/runs/0elf9wgr
